crates/bevy_ecs/src/schedule/executor/single_threaded.rs

```diff
@@ -44,13 +61,57 @@ impl SystemExecutor for SingleThreadedExecutor {
     }
 
     fn run(&mut self, schedule: &mut SystemSchedule, world: &mut World) {
+        /// return the next system not exempt from stepping starting at, and
+        /// inclusive of, first.  If there are no more systems that can be
+        /// stepped, the first non-exempt system index will be returned.
+        fn next_system(schedule: &SystemSchedule, first: usize) -> usize {
+            for i in first..schedule.systems_with_stepping.len() {
+                if schedule.systems_with_stepping[i] {
+                    return i;
+                }
+            }
+            for i in 0..schedule.systems_with_stepping.len() {
+                if schedule.systems_with_stepping[i] {
+                    return i;
+                }
+            }
+            panic!("all systems exempt from stepping");
```

Comment:
This can just be an `error` I think.

Reply:
Is this `error!` from `bevy::log::LogPlugin`?  I don't see any other executors making use of the log plugins.

I think the best approach right now is to remove the `panic!`, but add an `error!` call from `SystemSchedule::enable_stepping()`.

I'm not really clear on which areas of bevy it's safe/idiomatic to make use of `bevy::log`.  Is there a guideline for this?

Reply:
ahh, this is `tracing::error!`.  I did find it in use in `Schedule`.  Same plan, but will `use tracing::error`.

Reply:
Ok, so `tracing::error!` didn't display anything because tracing feature isn't enabled.

Back to my original question of can we use `bevy::log` from `bevy_ecs`?  This should be a visible warning to the user otherwise it's a lot of "why this no work?"

Reply:
Ok, nevermind.  Not gonna do this.  For the check to work, the schedule must have been run once already and internal data populated.  Instead I'll ensure the executor code can handle no systems stepping.

Reply:
I like that better!